### PR TITLE
update install mode with manual option

### PR DESCRIPTION
**Pull request by YCShen1010**
script enhancement for ticket: https://github.ibm.com/ibmprivatecloud/roadmap/issues/57080

### CODE REVIEW

File: cp3pt0-deployment/setup_singleton.sh [hunk 1]
```diff
@@ -218,6 +222,11 @@ function pre_req() {
         error "License not accepted. Rerun script with --license-accept flag set. See https://ibm.biz/integration-licenses for more details"
     fi
 
+    # Check INSTALL_MODE
+    if [[ "$INSTALL_MODE" != "Automatic" && "$INSTALL_MODE" != "Manual" ]]; then
+        error "Invalid INSTALL_MODE: $INSTALL_MODE, please try again"
```
Comment (Daniel-Fan) [inline cp3pt0-deployment/setup_singleton.sh:227]:
```suggestion
        error "Invalid INSTALL_MODE: $INSTALL_MODE. Valid values are 'Automatic' or 'Manual'"
```

File: cp3pt0-deployment/setup_tenant.sh [hunk 2]
```diff
@@ -155,6 +155,11 @@ function pre_req() {
         fi
     fi
 
+    # Check INSTALL_MODE
+    if [[ "$INSTALL_MODE" != "Automatic" && "$INSTALL_MODE" != "Manual" ]]; then
+        error "Invalid INSTALL_MODE: $INSTALL_MODE, please try again"
```
Comment (Daniel-Fan) [inline cp3pt0-deployment/setup_tenant.sh:160]:
```suggestion
        error "Invalid INSTALL_MODE: $INSTALL_MODE. Valid values are 'Automatic' or 'Manual'"
```

File: cp3pt0-deployment/migrate_tenant.sh [hunk 3]
```diff
@@ -240,6 +240,11 @@ function pre_req() {
         SOURCE_NS=$OPERATOR_NS
     fi
 
+    # Check INSTALL_MODE
+    if [[ "$INSTALL_MODE" != "Automatic" && "$INSTALL_MODE" != "Manual" ]]; then
+        error "Invalid INSTALL_MODE: $INSTALL_MODE, please try again"
```
Comment (Daniel-Fan) [inline cp3pt0-deployment/migrate_tenant.sh:245]:
```suggestion
        error "Invalid INSTALL_MODE: $INSTALL_MODE. Valid values are 'Automatic' or 'Manual'"
```

File: cp3pt0-deployment/common/utils.sh [hunk 4]
```diff
@@ -90,12 +90,16 @@ function wait_for_condition() {
...
     local error_message=$6
+    local install_mode=$7
 
     info "${wait_message}"
     while true; do
         result=$(eval "${condition}")
 
         if [[ ( ${retries} -eq 0 ) && ( -z "${result}" ) ]]; then
+            if [[ "${install_mode}" == "Manual" ]]; then
+                info "InstallPlan is not approved yet"
```
Comment (Daniel-Fan) [inline cp3pt0-deployment/common/utils.sh:101]:
```suggestion
                warning "InstallPlan is not approved yet"
```

File: cp3pt0-deployment/common/utils.sh [hunk 5]
```diff
@@ -104,6 +108,9 @@ function wait_for_condition() {
         
         if [[ -z "${result}" ]]; then
             info "RETRYING: ${wait_message} (${retries} left)"
+            if [[ "${install_mode}" == "Manual" ]]; then
+                info "Please manually approve installPlan to make upgrade proceeding..."
```
Comment (Daniel-Fan) [inline cp3pt0-deployment/common/utils.sh:112]:
```suggestion
                warning "Please manually approve installPlan to make upgrade proceeding..."
```

**Comment by YCShen1010** [comment]
output looks like:
```
[INFO] Waiting for operator ibm-common-service-operator to be upgraded 
Please manually approve installPlan to make upgrade proceeding...
[INFO] RETRYING: Waiting for operator ibm-common-service-operator to be upgraded 
Please manually approve installPlan to make upgrade proceeding... (3 left)
[INFO] RETRYING: Waiting for operator ibm-common-service-operator to be upgraded 
Please manually approve installPlan to make upgrade proceeding... (2 left)
[INFO] RETRYING: Waiting for operator ibm-common-service-operator to be upgraded 
Please manually approve installPlan to make upgrade proceeding... (1 left)
[✘] Timeout after 3 minutes waiting for operator ibm-common-service-operator to be upgraded 
InstallPlan is not manually approved yet
```

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**